eng/illink.targets

```diff
@@ -1,5 +1,23 @@
 <Project>
 
+  <PropertyGroup>
+    <_ILLinkTasksSourceDir>$(ToolsProjectRoot)illink/src/ILLink.Tasks/</_ILLinkTasksSourceDir>
+    <ILLinkAnalyzersPropsPath>$(ToolsProjectRoot)illink/src/ILLink.RoslynAnalyzer/build/Microsoft.NET.ILLink.Analyzers.props</ILLinkAnalyzersPropsPath>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- SkipGetTargetFrameworkProperties builds ILLink.Tasks without any TargetFramework settings
+         based on the referencing project, preventing errors when illink.targets is imported by a
+         project with TargetFrameworks that are incompatible with those of ILLink.Tasks. -->
+    <ProjectReference Include="$(_ILLinkTasksSourceDir)ILLink.Tasks.csproj"
+                      ReferenceOutputAssembly="false"
+                      PrivateAssets="all"
+                      SkipGetTargetFrameworkProperties="true"
+                      SetConfiguration="Configuration=$(ToolsConfiguration)" />
```

Comment:
Nearly there. Now that you skip retrieving target framework properties, you need to pass the target framework in, explicitly: `SetTargetFramework="TargetFramework=$(NetCoreAppToolCurrent)"` or `$(NetFrameworkToolCurrent)`. And add a comment to keep these in sync with https://github.com/dotnet/runtime/blob/9c3f8b3727d9be4de483a1d725c2bda22f956688/src/tools/illink/src/ILLink.Tasks/ILLink.Tasks.csproj#L4

You want to use the .NETCoreApp one when msbuild runs on .NETCoreApp, otherwise use the .NETFramework TFM.

```suggestion
    <!-- SkipGetTargetFrameworkProperties builds ILLink.Tasks without any TargetFramework settings
         based on the referencing project, preventing errors when illink.targets is imported by a
         project with TargetFrameworks that are incompatible with those of ILLink.Tasks. -->
    <ProjectReference Include="$(_ILLinkTasksSourceDir)ILLink.Tasks.csproj"
                      ReferenceOutputAssembly="false"
                      PrivateAssets="all"
                      SkipGetTargetFrameworkProperties="true"
                      SetConfiguration="Configuration=$(ToolsConfiguration)">
        <!-- Keep TFMs in sync with ILLink.Tasks.csproj -->
        <SetTargetFramework Condition="'$(MSBuildRuntimeType)' == 'Core'">TargetFramework=$(NetCoreAppToolCurrent)</SetTargetFramework>
        <SetTargetFramework Condition="'$(MSBuildRuntimeType)' != 'Core'">TargetFramework=$(NetFrameworkToolCurrent)</SetTargetFramework>
    </ProjectReference>
```